kernel/src/log_segment/tests.rs

```diff
@@ -620,3 +721,442 @@ fn table_changes_fails_with_larger_start_version_than_end() {
     let log_segment_res = LogSegment::for_table_changes(client.as_ref(), log_root, 1, Some(0));
     assert!(log_segment_res.is_err());
 }
+#[test]
+fn test_sidecar_to_filemeta_valid_paths() -> DeltaResult<()> {
+    let log_root = Url::parse("file:///var/_delta_log/")?;
+    let test_cases = [
+        (
+            "example.parquet",
+            "file:///var/_delta_log/_sidecars/example.parquet",
+        ),
+        (
+            "file:///var/_delta_log/_sidecars/example.parquet",
+            "file:///var/_delta_log/_sidecars/example.parquet",
+        ),
+        (
+            "test/test/example.parquet",
+            "file:///var/_delta_log/_sidecars/test/test/example.parquet",
+        ),
+    ];
+
+    for (input_path, expected_url) in test_cases.into_iter() {
+        let sidecar = Sidecar {
+            path: expected_url.to_string(),
+            modification_time: 0,
+            size_in_bytes: 1000,
+            tags: None,
+        };
+
+        let filemeta = LogSegment::sidecar_to_filemeta(&sidecar, &log_root)?;
+        assert_eq!(
+            filemeta.location.as_str(),
+            expected_url,
+            "Mismatch for input path: {}",
+            input_path
+        );
+    }
+    Ok(())
+}
+
```

Comment:
in the same vein of my comment above should we test that if user gives a schema with removes but no sidecar it should fail right?